### PR TITLE
dbcsr: Add latest releases and fix related issues

### PR DESCRIPTION
**Pull request by w8jcik**

### CODE REVIEW

**Review by dev-zero — DISMISSED** [review]

**Comment by dev-zero** [comment]
Not sure why testing is needed, though, can you elaborate please @w8jcik?

**Comment by w8jcik** [comment]
> Not sure why testing is needed, though, can you elaborate please @w8jcik?

I don't know the direct reason, but without this flag, CMake fails with following message

```
-- Found Python: /home/user/.spack/root/opt/spack/linux-ubuntu22.04-skylake/gcc-9.4.0/python-3.9.12-x4msfzae5cpz7r4ggq33caqlv72afasg/bin/python3.9 (found version "3.9.12") found components: Interpreter 
-- Found MPI_C: /home/user/.spack/root/opt/spack/linux-ubuntu22.04-skylake/gcc-9.4.0/openmpi-4.1.3-awkfuq5qz6p6oqwjnpa5trweop4iaika/lib/libmpi.so (found version "3.1") 
-- Found MPI_CXX: /home/user/.spack/root/opt/spack/linux-ubuntu22.04-skylake/gcc-9.4.0/openmpi-4.1.3-awkfuq5qz6p6oqwjnpa5trweop4iaika/lib/libmpi.so (found version "3.1") 
-- Found MPI_Fortran: /home/user/.spack/root/opt/spack/linux-ubuntu22.04-skylake/gcc-9.4.0/openmpi-4.1.3-awkfuq5qz6p6oqwjnpa5trweop4iaika/lib/libmpi_usempif08.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1") found components: C CXX Fortran 
-- Performing Test f2008-norm2
-- Performing Test f2008-norm2 - Success
-- Performing Test f2008-block_construct
-- Performing Test f2008-block_construct - Success
-- Performing Test f2008-contiguous
-- Performing Test f2008-contiguous - Success
-- Performing Test f95-reshape-order-allocatable
-- Performing Test f95-reshape-order-allocatable - Success
-- FYPP preprocessor found.
-- Configuring done
CMake Error at docs/CMakeLists.txt:17 (add_dependencies):
  The dependency target "doc_copy_tests" of target "doc" does not exist.


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

Looks like some hidden dependency.

**Comment by w8jcik** [comment]
@dev-zero The issue is quiet clear and here is a PR https://github.com/cp2k/dbcsr/pull/619

But we still need to handle releases 2.1 and 2.2. I have added a version-dependent fix. Is it fine?

**Comment by dev-zero** [comment]
@w8jcik perfectly, yes! Thanks a lot for the fix for DBCSR itself!

**Comment by w8jcik** [comment]
@dev-zero Thank you for merging the fix, but what with this Pull request? Are you concerned about something or can it be approved?

**Review by dev-zero — DISMISSED** [review]

**Comment by dev-zero** [comment]
sorry, I missed the dismissal of my previous review, all good from my side

File: var/spack/repos/builtin/packages/dbcsr/package.py [hunk 1]
```diff
@@ -33,7 +38,8 @@ class Dbcsr(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('mpi', when='+mpi')
     depends_on('libxsmm@1.11:~header-only', when='smm=libxsmm')
 
-    depends_on('cmake@3.12:', type='build')
+    depends_on('cmake@3.17:', type='build', when='@2.1:')
+    depends_on('cmake@3.10:', type='build', when='@:2.0')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/dbcsr/package.py]: I think it's easier to reason in terms of increasing requests with increasing versions:
```suggestion
    depends_on('cmake@3.10:', type='build')
```

**Review by alalazo — CHANGES_REQUESTED** [review]
One minor comment

**Review by alalazo — APPROVED** [review]

**Pull request merged**